src/mon/Monitor.cc

```diff
@@ -4534,7 +4544,12 @@ void Monitor::dispatch_op(MonOpRequestRef op)
       return;
     case MSG_COMMAND:
       op->set_type_command();
-      handle_tell_command(op);
+      try {
+        handle_tell_command(op);
+      } catch (const bad_cmd_get& e) {
+        reply_command(op, -EINVAL, e.what(), 0);
```

Comment:
`MMonCommandAck` the `reply_command()` generates is finally handled by `MonClient` which, as far as I see, doesn't use the version (`0` here) field.